docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_global_services_logging_duplicated.mdx

```diff
@@ -23,6 +23,299 @@ HIPAA
+
+By following these steps, you can effectively prevent duplicate entries in your CloudTrail logs using the AWS Management Console.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent duplicate entries in CloudTrail logs using AWS CLI, you can follow these steps:
+
+1. **Create a New CloudTrail Trail:**
+   Ensure you have a single, well-configured CloudTrail trail to avoid duplicate entries. Use the following command to create a new trail:
+
+   ```sh
+   aws cloudtrail create-trail --name my-trail --s3-bucket-name my-trail-bucket
+   ```
+
+2. **Enable Log File Validation:**
+   Enable log file validation to ensure the integrity of the log files and prevent duplicates. Use the following command:
+
+   ```sh
+   aws cloudtrail update-trail --name my-trail --enable-log-file-validation
+   ```
+
+3. **Ensure Only One Trail is Logging for Each Region:**
+   Verify that you have only one trail logging events for each region. List all trails and check their configurations:
+
+   ```sh
+   aws cloudtrail describe-trails
+   ```
+
+   If you find multiple trails logging for the same region, consider consolidating them.
+
+4. **Enable Multi-Region Trail:**
+   If you need to log events from multiple regions, enable the multi-region trail option to avoid creating separate trails for each region:
+
+   ```sh
+   aws cloudtrail update-trail --name my-trail --is-multi-region-trail
+   ```
+
+By following these steps, you can prevent duplicate entries in CloudTrail logs using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent duplicate entries in AWS CloudTrail logs using Python scripts, you can follow these steps:
+
+### 1. Set Up AWS SDK for Python (Boto3)
+First, ensure you have the AWS SDK for Python (Boto3) installed. You can install it using pip if you haven't already:
+
+```bash
+pip install boto3
+```
+
+### 2. Create a Python Script to Check for Duplicate Entries
+You can create a Python script that will check for duplicate entries in CloudTrail logs and take appropriate actions to prevent them.
+
+### 3. Use Boto3 to Interact with CloudTrail
+Here is a sample Python script that demonstrates how to interact with CloudTrail and check for duplicate entries:
+
+```python
+import boto3
+import hashlib
+
+# Initialize a session using Amazon CloudTrail
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+
+cloudtrail = session.client('cloudtrail')
+
+# Function to get CloudTrail events
+def get_cloudtrail_events():
+    response = cloudtrail.lookup_events(
+        LookupAttributes=[
+            {
+                'AttributeKey': 'EventName',
+                'AttributeValue': 'PutObject'
+            },
+        ],
+        MaxResults=50
+    )
+    return response['Events']
+
+# Function to check for duplicate entries
+def check_for_duplicates(events):
+    event_hashes = set()
+    duplicates = []
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>